logstash-core/src/main/java/org/logstash/plugins/ContextImpl.java

```diff
@@ -68,19 +64,6 @@ public DeprecationLogger getDeprecationLogger(Plugin plugin) {
 
     @Override
     public EventFactory getEventFactory() {
-        return new EventFactory() {
-            @Override
-            public Event newEvent() {
-                return new org.logstash.Event();
-            }
-
-            @Override
-            public Event newEvent(Map<? extends Serializable, Object> data) {
-                if (data instanceof ConvertedMap) {
-                    return new org.logstash.Event((ConvertedMap)data);
-                }
-                return new org.logstash.Event(data);
-            }
-        };
+        return BasicEventFactory.INSTANCE;
```

Comment:
I can see why adding an `EventFactory getTargetedEventFactory()` for the Java API is out of scope here -- the context doesn't have the plugin's config, and we don't have standardized ways of defining plugin options.

A native plugin could also _wrap_ the `EventFactory` that they retrieve from here, so it isn't that big of a deal anyway.

But we should file a follow-up task to make sure that we aren't leaving the Java API behind.